// xpring-common-js/build/generated/rpc/v1/xrp_ledger_grpc_web_pb.js
/**
 * @fileoverview gRPC-Web generated client stub for rpc.v1
 * @enhanceable
 * @public
 */

// GENERATED CODE -- DO NOT EDIT!



const grpc = {};
grpc.web = require('grpc-web');


var rpc_v1_account_info_pb = require('../../rpc/v1/account_info_pb.js')

var rpc_v1_fee_pb = require('../../rpc/v1/fee_pb.js')

var rpc_v1_submit_pb = require('../../rpc/v1/submit_pb.js')

var rpc_v1_tx_pb = require('../../rpc/v1/tx_pb.js')
const proto = {};
proto.rpc = {};
proto.rpc.v1 = require('./xrp_ledger_pb.js');

/**
 * @param {string} hostname
 * @param {?Object} credentials
 * @param {?Object} options
 * @constructor
 * @struct
 * @final
 */
proto.rpc.v1.XRPLedgerAPIServiceClient =
    function(hostname, credentials, options) {
  if (!options) options = {};
  options['format'] = 'text';

  /**
   * @private @const {!grpc.web.GrpcWebClientBase} The client
   */
  this.client_ = new grpc.web.GrpcWebClientBase(options);

  /**
   * @private @const {string} The hostname
   */
  this.hostname_ = hostname;

};


/**
 * @param {string} hostname
 * @param {?Object} credentials
 * @param {?Object} options
 * @constructor
 * @struct
 * @final
 */
proto.rpc.v1.XRPLedgerAPIServicePromiseClient =
    function(hostname, credentials, options) {
  if (!options) options = {};
  options['format'] = 'text';

  /**
   * @private @const {!grpc.web.GrpcWebClientBase} The client
   */
  this.client_ = new grpc.web.GrpcWebClientBase(options);

  /**
   * @private @const {string} The hostname
   */
  this.hostname_ = hostname;

};


/**
 * @const
 * @type {!grpc.web.MethodDescriptor<
 *   !proto.rpc.v1.GetAccountInfoRequest,
 *   !proto.rpc.v1.GetAccountInfoResponse>}
 */
const methodDescriptor_XRPLedgerAPIService_GetAccountInfo = new grpc.web.MethodDescriptor(
  '/rpc.v1.XRPLedgerAPIService/GetAccountInfo',
  grpc.web.MethodType.UNARY,
  rpc_v1_account_info_pb.GetAccountInfoRequest,
  rpc_v1_account_info_pb.GetAccountInfoResponse,
  /**
   * @param {!proto.rpc.v1.GetAccountInfoRequest} request
   * @return {!Uint8Array}
   */
  function(request) {
    return request.serializeBinary();
  },
  rpc_v1_account_info_pb.GetAccountInfoResponse.deserializeBinary
);


/**
 * @const
 * @type {!grpc.web.AbstractClientBase.MethodInfo<
 *   !proto.rpc.v1.GetAccountInfoRequest,
 *   !proto.rpc.v1.GetAccountInfoResponse>}
 */
const methodInfo_XRPLedgerAPIService_GetAccountInfo = new grpc.web.AbstractClientBase.MethodInfo(
  rpc_v1_account_info_pb.GetAccountInfoResponse,
  /**
   * @param {!proto.rpc.v1.GetAccountInfoRequest} request
   * @return {!Uint8Array}
   */
  function(request) {
    return request.serializeBinary();
  },
  rpc_v1_account_info_pb.GetAccountInfoResponse.deserializeBinary
);


/**
 * @param {!proto.rpc.v1.GetAccountInfoRequest} request The
 *     request proto
 * @param {?Object<string, string>} metadata User defined
 *     call metadata
 * @param {function(?grpc.web.Error, ?proto.rpc.v1.GetAccountInfoResponse)}
 *     callback The callback function(error, response)
 * @return {!grpc.web.ClientReadableStream<!proto.rpc.v1.GetAccountInfoResponse>|undefined}
 *     The XHR Node Readable Stream
 */
proto.rpc.v1.XRPLedgerAPIServiceClient.prototype.getAccountInfo =
    function(request, metadata, callback) {
  return this.client_.rpcCall(this.hostname_ +
      '/rpc.v1.XRPLedgerAPIService/GetAccountInfo',
      request,
      metadata || {},
      methodDescriptor_XRPLedgerAPIService_GetAccountInfo,
      callback);
};


/**
 * @param {!proto.rpc.v1.GetAccountInfoRequest} request The
 *     request proto
 * @param {?Object<string, string>} metadata User defined
 *     call metadata
 * @return {!Promise<!proto.rpc.v1.GetAccountInfoResponse>}
 *     A native promise that resolves to the response
 */
proto.rpc.v1.XRPLedgerAPIServicePromiseClient.prototype.getAccountInfo =
    function(request, metadata) {
  return this.client_.unaryCall(this.hostname_ +
      '/rpc.v1.XRPLedgerAPIService/GetAccountInfo',
      request,
      metadata || {},
      methodDescriptor_XRPLedgerAPIService_GetAccountInfo);
};


/**
 * @const
 * @type {!grpc.web.MethodDescriptor<
 *   !proto.rpc.v1.GetFeeRequest,
 *   !proto.rpc.v1.GetFeeResponse>}
 */
const methodDescriptor_XRPLedgerAPIService_GetFee = new grpc.web.MethodDescriptor(
  '/rpc.v1.XRPLedgerAPIService/GetFee',
  grpc.web.MethodType.UNARY,
  rpc_v1_fee_pb.GetFeeRequest,
  rpc_v1_fee_pb.GetFeeResponse,
  /**
   * @param {!proto.rpc.v1.GetFeeRequest} request
   * @return {!Uint8Array}
   */
  function(request) {
    return request.serializeBinary();
  },
  rpc_v1_fee_pb.GetFeeResponse.deserializeBinary
);


/**
 * @const
 * @type {!grpc.web.AbstractClientBase.MethodInfo<
 *   !proto.rpc.v1.GetFeeRequest,
 *   !proto.rpc.v1.GetFeeResponse>}
 */
const methodInfo_XRPLedgerAPIService_GetFee = new grpc.web.AbstractClientBase.MethodInfo(
  rpc_v1_fee_pb.GetFeeResponse,
  /**
   * @param {!proto.rpc.v1.GetFeeRequest} request
   * @return {!Uint8Array}
   */
  function(request) {
    return request.serializeBinary();
  },
  rpc_v1_fee_pb.GetFeeResponse.deserializeBinary
);


/**
 * @param {!proto.rpc.v1.GetFeeRequest} request The
 *     request proto
 * @param {?Object<string, string>} metadata User defined
 *     call metadata
 * @param {function(?grpc.web.Error, ?proto.rpc.v1.GetFeeResponse)}
 *     callback The callback function(error, response)
 * @return {!grpc.web.ClientReadableStream<!proto.rpc.v1.GetFeeResponse>|undefined}
 *     The XHR Node Readable Stream
 */
proto.rpc.v1.XRPLedgerAPIServiceClient.prototype.getFee =
    function(request, metadata, callback) {
  return this.client_.rpcCall(this.hostname_ +
      '/rpc.v1.XRPLedgerAPIService/GetFee',
      request,
      metadata || {},
      methodDescriptor_XRPLedgerAPIService_GetFee,
      callback);
};


/**
 * @param {!proto.rpc.v1.GetFeeRequest} request The
 *     request proto
 * @param {?Object<string, string>} metadata User defined
 *     call metadata
 * @return {!Promise<!proto.rpc.v1.GetFeeResponse>}
 *     A native promise that resolves to the response
 */
proto.rpc.v1.XRPLedgerAPIServicePromiseClient.prototype.getFee =
    function(request, metadata) {
  return this.client_.unaryCall(this.hostname_ +
      '/rpc.v1.XRPLedgerAPIService/GetFee',
      request,
      metadata || {},
      methodDescriptor_XRPLedgerAPIService_GetFee);
};


/**
 * @const
 * @type {!grpc.web.MethodDescriptor<
 *   !proto.rpc.v1.SubmitTransactionRequest,
 *   !proto.rpc.v1.SubmitTransactionResponse>}
 */
const methodDescriptor_XRPLedgerAPIService_SubmitTransaction = new grpc.web.MethodDescriptor(
  '/rpc.v1.XRPLedgerAPIService/SubmitTransaction',
  grpc.web.MethodType.UNARY,
  rpc_v1_submit_pb.SubmitTransactionRequest,
  rpc_v1_submit_pb.SubmitTransactionResponse,
  /**
   * @param {!proto.rpc.v1.SubmitTransactionRequest} request
   * @return {!Uint8Array}
   */
  function(request) {
    return request.serializeBinary();
  },
  rpc_v1_submit_pb.SubmitTransactionResponse.deserializeBinary
);


/**
 * @const
 * @type {!grpc.web.AbstractClientBase.MethodInfo<
 *   !proto.rpc.v1.SubmitTransactionRequest,
 *   !proto.rpc.v1.SubmitTransactionResponse>}
 */
const methodInfo_XRPLedgerAPIService_SubmitTransaction = new grpc.web.AbstractClientBase.MethodInfo(
  rpc_v1_submit_pb.SubmitTransactionResponse,
  /**
   * @param {!proto.rpc.v1.SubmitTransactionRequest} request
   * @return {!Uint8Array}
   */
  function(request) {
    return request.serializeBinary();
  },
  rpc_v1_submit_pb.SubmitTransactionResponse.deserializeBinary
);


/**
 * @param {!proto.rpc.v1.SubmitTransactionRequest} request The
 *     request proto
 * @param {?Object<string, string>} metadata User defined
 *     call metadata
 * @param {function(?grpc.web.Error, ?proto.rpc.v1.SubmitTransactionResponse)}
 *     callback The callback function(error, response)
 * @return {!grpc.web.ClientReadableStream<!proto.rpc.v1.SubmitTransactionResponse>|undefined}
 *     The XHR Node Readable Stream
 */
proto.rpc.v1.XRPLedgerAPIServiceClient.prototype.submitTransaction =
    function(request, metadata, callback) {
  return this.client_.rpcCall(this.hostname_ +
      '/rpc.v1.XRPLedgerAPIService/SubmitTransaction',
      request,
      metadata || {},
      methodDescriptor_XRPLedgerAPIService_SubmitTransaction,
      callback);
};


/**
 * @param {!proto.rpc.v1.SubmitTransactionRequest} request The
 *     request proto
 * @param {?Object<string, string>} metadata User defined
 *     call metadata
 * @return {!Promise<!proto.rpc.v1.SubmitTransactionResponse>}
 *     A native promise that resolves to the response
 */
proto.rpc.v1.XRPLedgerAPIServicePromiseClient.prototype.submitTransaction =
    function(request, metadata) {
  return this.client_.unaryCall(this.hostname_ +
      '/rpc.v1.XRPLedgerAPIService/SubmitTransaction',
      request,
      metadata || {},
      methodDescriptor_XRPLedgerAPIService_SubmitTransaction);
};


/**
 * @const
 * @type {!grpc.web.MethodDescriptor<
 *   !proto.rpc.v1.GetTxRequest,
 *   !proto.rpc.v1.GetTxResponse>}
 */
const methodDescriptor_XRPLedgerAPIService_GetTx = new grpc.web.MethodDescriptor(
  '/rpc.v1.XRPLedgerAPIService/GetTx',
  grpc.web.MethodType.UNARY,
  rpc_v1_tx_pb.GetTxRequest,
  rpc_v1_tx_pb.GetTxResponse,
  /**
   * @param {!proto.rpc.v1.GetTxRequest} request
   * @return {!Uint8Array}
   */
  function(request) {
    return request.serializeBinary();
  },
  rpc_v1_tx_pb.GetTxResponse.deserializeBinary
);


/**
 * @const
 * @type {!grpc.web.AbstractClientBase.MethodInfo<
 *   !proto.rpc.v1.GetTxRequest,
 *   !proto.rpc.v1.GetTxResponse>}
 */
const methodInfo_XRPLedgerAPIService_GetTx = new grpc.web.AbstractClientBase.MethodInfo(
  rpc_v1_tx_pb.GetTxResponse,
  /**
   * @param {!proto.rpc.v1.GetTxRequest} request
   * @return {!Uint8Array}
   */
  function(request) {
    return request.serializeBinary();
  },
  rpc_v1_tx_pb.GetTxResponse.deserializeBinary
);


/**
 * @param {!proto.rpc.v1.GetTxRequest} request The
 *     request proto
 * @param {?Object<string, string>} metadata User defined
 *     call metadata
 * @param {function(?grpc.web.Error, ?proto.rpc.v1.GetTxResponse)}
 *     callback The callback function(error, response)
 * @return {!grpc.web.ClientReadableStream<!proto.rpc.v1.GetTxResponse>|undefined}
 *     The XHR Node Readable Stream
 */
proto.rpc.v1.XRPLedgerAPIServiceClient.prototype.getTx =
    function(request, metadata, callback) {
  return this.client_.rpcCall(this.hostname_ +
      '/rpc.v1.XRPLedgerAPIService/GetTx',
      request,
      metadata || {},
      methodDescriptor_XRPLedgerAPIService_GetTx,
      callback);
};


/**
 * @param {!proto.rpc.v1.GetTxRequest} request The
 *     request proto
 * @param {?Object<string, string>} metadata User defined
 *     call metadata
 * @return {!Promise<!proto.rpc.v1.GetTxResponse>}
 *     A native promise that resolves to the response
 */
proto.rpc.v1.XRPLedgerAPIServicePromiseClient.prototype.getTx =
    function(request, metadata) {
  return this.client_.unaryCall(this.hostname_ +
      '/rpc.v1.XRPLedgerAPIService/GetTx',
      request,
      metadata || {},
      methodDescriptor_XRPLedgerAPIService_GetTx);
};


module.exports = proto.rpc.v1;
