// xpring-js/build/generated/node/rpc/v1/xrp_ledger_grpc_pb.js
// GENERATED CODE -- DO NOT EDIT!

'use strict';
var grpc = require('grpc');
var rpc_v1_account_info_pb = require('../../rpc/v1/account_info_pb.js');
var rpc_v1_fee_pb = require('../../rpc/v1/fee_pb.js');
var rpc_v1_submit_pb = require('../../rpc/v1/submit_pb.js');
var rpc_v1_tx_pb = require('../../rpc/v1/tx_pb.js');

function serialize_rpc_v1_GetAccountInfoRequest(arg) {
  if (!(arg instanceof rpc_v1_account_info_pb.GetAccountInfoRequest)) {
    throw new Error('Expected argument of type rpc.v1.GetAccountInfoRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_rpc_v1_GetAccountInfoRequest(buffer_arg) {
  return rpc_v1_account_info_pb.GetAccountInfoRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_rpc_v1_GetAccountInfoResponse(arg) {
  if (!(arg instanceof rpc_v1_account_info_pb.GetAccountInfoResponse)) {
    throw new Error('Expected argument of type rpc.v1.GetAccountInfoResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_rpc_v1_GetAccountInfoResponse(buffer_arg) {
  return rpc_v1_account_info_pb.GetAccountInfoResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_rpc_v1_GetFeeRequest(arg) {
  if (!(arg instanceof rpc_v1_fee_pb.GetFeeRequest)) {
    throw new Error('Expected argument of type rpc.v1.GetFeeRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_rpc_v1_GetFeeRequest(buffer_arg) {
  return rpc_v1_fee_pb.GetFeeRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_rpc_v1_GetFeeResponse(arg) {
  if (!(arg instanceof rpc_v1_fee_pb.GetFeeResponse)) {
    throw new Error('Expected argument of type rpc.v1.GetFeeResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_rpc_v1_GetFeeResponse(buffer_arg) {
  return rpc_v1_fee_pb.GetFeeResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_rpc_v1_GetTxRequest(arg) {
  if (!(arg instanceof rpc_v1_tx_pb.GetTxRequest)) {
    throw new Error('Expected argument of type rpc.v1.GetTxRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_rpc_v1_GetTxRequest(buffer_arg) {
  return rpc_v1_tx_pb.GetTxRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_rpc_v1_GetTxResponse(arg) {
  if (!(arg instanceof rpc_v1_tx_pb.GetTxResponse)) {
    throw new Error('Expected argument of type rpc.v1.GetTxResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_rpc_v1_GetTxResponse(buffer_arg) {
  return rpc_v1_tx_pb.GetTxResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_rpc_v1_SubmitTransactionRequest(arg) {
  if (!(arg instanceof rpc_v1_submit_pb.SubmitTransactionRequest)) {
    throw new Error('Expected argument of type rpc.v1.SubmitTransactionRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_rpc_v1_SubmitTransactionRequest(buffer_arg) {
  return rpc_v1_submit_pb.SubmitTransactionRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_rpc_v1_SubmitTransactionResponse(arg) {
  if (!(arg instanceof rpc_v1_submit_pb.SubmitTransactionResponse)) {
    throw new Error('Expected argument of type rpc.v1.SubmitTransactionResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_rpc_v1_SubmitTransactionResponse(buffer_arg) {
  return rpc_v1_submit_pb.SubmitTransactionResponse.deserializeBinary(new Uint8Array(buffer_arg));
}


// RPCs available to interact with the XRP Ledger.
var XRPLedgerAPIServiceService = exports.XRPLedgerAPIServiceService = {
  // Get account info for an account on the XRP Ledger.
getAccountInfo: {
    path: '/rpc.v1.XRPLedgerAPIService/GetAccountInfo',
    requestStream: false,
    responseStream: false,
    requestType: rpc_v1_account_info_pb.GetAccountInfoRequest,
    responseType: rpc_v1_account_info_pb.GetAccountInfoResponse,
    requestSerialize: serialize_rpc_v1_GetAccountInfoRequest,
    requestDeserialize: deserialize_rpc_v1_GetAccountInfoRequest,
    responseSerialize: serialize_rpc_v1_GetAccountInfoResponse,
    responseDeserialize: deserialize_rpc_v1_GetAccountInfoResponse,
  },
  // Get the fee for a transaction on the XRP Ledger.
getFee: {
    path: '/rpc.v1.XRPLedgerAPIService/GetFee',
    requestStream: false,
    responseStream: false,
    requestType: rpc_v1_fee_pb.GetFeeRequest,
    responseType: rpc_v1_fee_pb.GetFeeResponse,
    requestSerialize: serialize_rpc_v1_GetFeeRequest,
    requestDeserialize: deserialize_rpc_v1_GetFeeRequest,
    responseSerialize: serialize_rpc_v1_GetFeeResponse,
    responseDeserialize: deserialize_rpc_v1_GetFeeResponse,
  },
  // Submit a signed transaction to the XRP Ledger.
submitTransaction: {
    path: '/rpc.v1.XRPLedgerAPIService/SubmitTransaction',
    requestStream: false,
    responseStream: false,
    requestType: rpc_v1_submit_pb.SubmitTransactionRequest,
    responseType: rpc_v1_submit_pb.SubmitTransactionResponse,
    requestSerialize: serialize_rpc_v1_SubmitTransactionRequest,
    requestDeserialize: deserialize_rpc_v1_SubmitTransactionRequest,
    responseSerialize: serialize_rpc_v1_SubmitTransactionResponse,
    responseDeserialize: deserialize_rpc_v1_SubmitTransactionResponse,
  },
  // Get the status of a transaction
getTx: {
    path: '/rpc.v1.XRPLedgerAPIService/GetTx',
    requestStream: false,
    responseStream: false,
    requestType: rpc_v1_tx_pb.GetTxRequest,
    responseType: rpc_v1_tx_pb.GetTxResponse,
    requestSerialize: serialize_rpc_v1_GetTxRequest,
    requestDeserialize: deserialize_rpc_v1_GetTxRequest,
    responseSerialize: serialize_rpc_v1_GetTxResponse,
    responseDeserialize: deserialize_rpc_v1_GetTxResponse,
  },
};

exports.XRPLedgerAPIServiceClient = grpc.makeGenericClientConstructor(XRPLedgerAPIServiceService);
